src/Sulu/Bundle/ResourceBundle/Filter/FilterManager.php

```diff
@@ -0,0 +1,385 @@
+<?php
+/*
+ * This file is part of the Sulu CMS.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ResourceBundle\Filter;
+
+use Doctrine\ORM\EntityManager;
+use Sulu\Bundle\ResourceBundle\Api\ConditionGroup;
+use Sulu\Bundle\ResourceBundle\Api\Condition;
+use Sulu\Bundle\ResourceBundle\Entity\Condition as ConditionEntity;
+use Sulu\Bundle\ResourceBundle\Entity\ConditionGroup as ConditionGroupEntity;
+use Sulu\Bundle\ResourceBundle\Api\Filter;
+use Sulu\Bundle\ResourceBundle\Entity\ConditionGroupRepositoryInterface;
+use Sulu\Bundle\ResourceBundle\Entity\Filter as FilterEntity;
+use Sulu\Bundle\ResourceBundle\Entity\FilterRepositoryInterface;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\FilterDependencyNotFoundException;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\FilterNotFoundException;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\MissingConditionAttributeException;
+use Sulu\Bundle\ResourceBundle\Filter\Exception\MissingFilterAttributeException;
+use Sulu\Component\Persistence\RelationTrait;
+use Sulu\Component\Rest\Exception\EntityIdAlreadySetException;
+use Sulu\Component\Rest\ListBuilder\Doctrine\FieldDescriptor\DoctrineFieldDescriptor;
+use Sulu\Component\Rest\ListBuilder\Doctrine\FieldDescriptor\DoctrineJoinDescriptor;
+use Sulu\Component\Security\Authentication\UserRepositoryInterface;
+
+/**
+ * Class FilterManager
+ * @package Sulu\Bundle\ResourceBundle\Filter
+ */
+class FilterManager implements FilterManagerInterface
+{
+    use RelationTrait;
+
+    protected static $filterEntityName = 'SuluResourceBundle:Filter';
+    protected static $conditionGroupEntityName = 'SuluResourceBundle:ConditionGroup';
+    protected static $conditionEntityName = 'SuluResourceBundle:Condition';
+    protected static $filterTranslationEntityName = 'SuluResourceBundle:FilterTranslation';
+
+    /**
+     * @var EntityManager
+     */
+    protected $em;
+
+    /**
+     * @var FilterRepositoryInterface
+     */
+    protected $filterRepository;
+
+    /**
+     * @var UserRepositoryInterface
+     */
+    protected $userRepository;
+
+    /**
+     * @var ConditionGroupRepositoryInterface
+     */
+    protected $conditionGroupRepository;
+
+    public function __construct(
+        $em,
```

Comment:
tpyehint missing! do you need the entitymanager interface or is the objectmanager enough?